src/hooks/useSingleExecution/index.ts

```diff
@@ -1,22 +1,20 @@
-import {useCallback} from 'react';
-
 type Action<T extends unknown[]> = (...params: T) => void | Promise<void>;
 
+const singleExecution =
+    <T extends unknown[]>(action?: Action<T>) =>
+    (...params: T) => {
+        action?.(...params);
+    };
+
+const hookValue = {isExecuting: false, singleExecution};
+
 /**
  * This hook was specifically written for native issue
  * more information: https://github.com/Expensify/App/pull/24614 https://github.com/Expensify/App/pull/24173
  * on web we don't need this mechanism so we just call the action directly.
  */
 export default function useSingleExecution() {
-    const singleExecution = useCallback(
```

Comment:
I wonder if we could create a lint rule that catches any "pure function" declared inside a hook or component, and force us to move the function declaration outside of the component.

Reply:
Yeah that's an interesting topic. It might be quite tricky to achieve - rule would need deeper understanding of current lexical scope. I will search for it tho!